internal/controllers/storage/init.go

```diff
@@ -101,88 +107,243 @@ func (r *Reconciler) setInitStorageCompleted(
 		)
 		return Stop, ctrl.Result{RequeueAfter: DefaultRequeueDelay}, err
 	}
 
-	cmd := []string{
-		fmt.Sprintf("%s/%s", v1alpha1.BinariesDir, v1alpha1.DaemonBinaryName),
-	}
-
-	if storage.Spec.OperatorConnection != nil {
-		ydbCtx, cancel := context.WithTimeout(ctx, time.Second)
-		defer cancel()
-		token, err := creds.Token(
-			metadata.AppendToOutgoingContext(ydbCtx, "x-ydb-database", storage.Spec.Domain),
-		)
+	if initJob.Status.Succeeded > 0 {
+		r.Log.Info("Init Job status succeeded")
+		podLogs, err := r.getSucceededJobLogs(ctx, storage, initJob)
 		if err != nil {
-			r.Log.Error(err, "initializeStorage error")
-			return Stop, ctrl.Result{RequeueAfter: StorageInitializationRequeueDelay}, err
+			r.Recorder.Event(
+				storage,
+				corev1.EventTypeWarning,
+				"ControllerError",
+				fmt.Sprintf("Failed to get succeeded Pod for Job: %s", err),
+			)
+			return Stop, ctrl.Result{RequeueAfter: DefaultRequeueDelay}, err
 		}
-		cmd = append(
-			cmd,
-			"--token",
-			token,
-		)
-	}
-
-	cmd = append(
-		cmd,
-		"-s",
-		storage.GetStorageEndpointWithProto(),
-	)
 
-	cmd = append(
-		cmd,
-		"admin", "blobstorage", "config", "init",
-		"--yaml-file",
-		fmt.Sprintf("%s/%s", v1alpha1.ConfigDir, v1alpha1.ConfigFileName),
-	)
-
-	stdout, _, err := exec.InPod(r.Scheme, r.Config, storage.Namespace, podList.Items[0].Name, "ydb-storage", cmd)
-	if err != nil {
-		if mismatchItemConfigGenerationRegexp.MatchString(stdout) {
+		if mismatchItemConfigGenerationRegexp.MatchString(podLogs) {
 			r.Log.Info("Storage is already initialized, continuing...")
 			r.Recorder.Event(
 				storage,
 				corev1.EventTypeNormal,
 				"InitializingStorage",
 				"Storage initialization attempted and skipped, storage already initialized",
 			)
-			return r.setInitStorageCompleted(
-				ctx,
+			return r.setInitStorageCompleted(ctx, storage, "Storage already initialized")
+		}
+
+		r.Recorder.Event(
+			storage,
+			corev1.EventTypeNormal,
+			"InitializingStorage",
+			"Storage initialized successfully",
+		)
+		return r.setInitStorageCompleted(ctx, storage, "Storage initialized successfully")
+	}
+
+	if initJob.Status.Failed > 0 {
+		r.Log.Info("Init Job status failed")
+		r.Recorder.Event(
+			storage,
+			corev1.EventTypeWarning,
+			"InitializingStorage",
+			"Failed to initializing Storage",
+		)
+		if err := r.Delete(ctx, initJob); err != nil {
+			r.Recorder.Event(
 				storage,
-				"Storage already initialized",
+				corev1.EventTypeWarning,
+				"ControllerError",
+				fmt.Sprintf("Failed to delete Job: %s", err),
 			)
+			return Stop, ctrl.Result{RequeueAfter: DefaultRequeueDelay}, err
 		}
+	}
+
+	return Stop, ctrl.Result{RequeueAfter: StorageInitializationRequeueDelay}, nil
+}
 
-		return Stop, ctrl.Result{RequeueAfter: StorageInitializationRequeueDelay}, err
+func (r *Reconciler) getSucceededJobLogs(
+	ctx context.Context,
+	storage *resources.StorageClusterBuilder,
+	job *batchv1.Job,
+) (string, error) {
+	podList := &corev1.PodList{}
```

Comment:
handleErr := func(err error) (string, error) {
return "", fmt.Errorf("package.blabla. getSucceededJobLogs: %w", err)
}

....

return handleErr(fmt.Errorf("my context: blabla: %w", err))


up to you